### PR TITLE
[feature/otx] Update initial doc template

### PR DESCRIPTION
**Pull request by chuneuny-emily**

### CODE REVIEW

File: docs/source/get_started/index.rst [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Get Started
```
Comment (samet-akcay) [inline docs/source/get_started/index.rst:1]: Getting started, maybe?

**Review by samet-akcay — COMMENTED** [review]
@chuneuny-emily, thank you for creating this PR.

The initial doc structure was inspired by [this website](https://documentation.divio.com/) and [PyCon talk](https://www.youtube.com/watch?v=t4vKPhjcMZg).

I have couple of questions based on the changes:

- `explanation` directory has been replaced with `design` directory. The initial idea was to include any sort of documentation that explains the theoretical foundations, including design principles and some algorithmic background. Do you think `design` directory would cover algorithmic background. Or do you think we wouldn't need to include algo background but just mention the design patterns here?
- I see that `how-to-guides` is replaced with `additional-guides`. I wasn't hundred percent sure about `how-to-guides` anyway since it is not clearly distinguishable from `tutorials`. I'm now wondering whether it would be an idea to have `tutorials` section, which contains `beginner`, `intermediate` and `advanced` tutorials. We could place the `addiitonal-guides` to either `intermediate` or `advanced` tutorials. This way, user would know how to navigate between the `tutorials` and `additional-guides`. Also, the technical level of the tutorial would become more apparent. Any thoughts?
- I am not sure if I understand the role of `samples` directory. Could you please elaborate?

File: docs/source/tutorials/installation.rst [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+Python's ``virtualenv`` and ``venv``.
+
+Install OTX
+-----------
+The library can be installed via either PyPI or editable local install.
+
+Local Install
+~~~~~~~~~~~~~
+Local install is currently the best approach to install the library since it
+will also allow the user to edit the changes within the codebase.
```
Comment (kprokofi) [inline docs/source/tutorials/installation.rst]:
```suggestion
will allow the user to edit the changes within the codebase.
```

File: docs/source/tutorials/installation.rst [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+could install this option.
+
+.. code-block:: bash
+
+    # Install API and CLI only.
+    pip install otx
+
+Classification Task Installation
+""""""""""""""""""""""""""""""""
+Classification option would install ``otx`` with classification algorithms only.
```
Comment (kprokofi) [inline docs/source/tutorials/installation.rst]: Maybe it is worth noticing that there are 3 different classification task types. Also, all other tasks below have a description of the source library that the algorithm utilizes, but for classification, there is no such explanation. I think we should add it.

File: docs/source/tutorials/installation.rst [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+.. code-block:: bash
+
+    # Install otx anomaly library
+    pip install otx[anomaly]
+
+Full Installation
+"""""""""""""""""
+``full``` option would install all of the tasks together, overall making ``otx``
+a complete library that a user could train a supervised, semi-supervised or
+self-supervised classification, detection or segmentation algorithms with full
```
Comment (kprokofi) [inline docs/source/tutorials/installation.rst]: self-supervised is not intended to be added in Q4 and early Q1 at least. Should we mention this? Or is it just something global for the future?

**Comment by kprokofi** [comment]
> @chuneuny-emily, thank you for creating this PR.
> 
> The initial doc structure was inspired by [this website](https://documentation.divio.com/) and [PyCon talk](https://www.youtube.com/watch?v=t4vKPhjcMZg).
> 
> I have couple of questions based on the changes:
> 
> * `explanation` directory has been replaced with `design` directory. The initial idea was to include any sort of documentation that explains the theoretical foundations, including design principles and some algorithmic background. Do you think `design` directory would cover algorithmic background. Or do you think we wouldn't need to include algo background but just mention the design patterns here?
> * I see that `how-to-guides` is replaced with `additional-guides`. I wasn't hundred percent sure about `how-to-guides` anyway since it is not clearly distinguishable from `tutorials`. I'm now wondering whether it would be an idea to have `tutorials` section, which contains `beginner`, `intermediate` and `advanced` tutorials. We could place the `addiitonal-guides` to either `intermediate` or `advanced` tutorials. This way, user would know how to navigate between the `tutorials` and `additional-guides`. Also, the technical level of the tutorial would become more apparent. Any thoughts?
> * I am not sure if I understand the role of `samples` directory. Could you please elaborate?

@chuneuny-emily , joining Samet, I would like to understand which part of the documentation is responsible for what ideologically. Could you describe a little more about each section? Thank you.

File: docs/source/get_started/index.rst (hunk 1 above)
Comment (chuneuny-emily) [reply docs/source/get_started/index.rst:1]: I thought like it, but Geti used the expression like "GET STARTED" at https://sc-demo.iotg.sclab.intel.com/docs/guide/get-started/supported-tasks.html, which was shared by documentation team.

File: docs/source/tutorials/installation.rst (hunk 2 above)
Comment (chuneuny-emily) [reply docs/source/tutorials/installation.rst]: Yes. Looks good correction.

File: docs/source/tutorials/installation.rst (hunk 3 above)
Comment (chuneuny-emily) [reply docs/source/tutorials/installation.rst]: Yes. Right.

File: docs/source/tutorials/installation.rst (hunk 4 above)
Comment (chuneuny-emily) [reply docs/source/tutorials/installation.rst]: I think we should mention it. e.g. self-supervised classification(planned to be added in Q2'23)

**Comment by chuneuny-emily** [comment]
> @chuneuny-emily, thank you for creating this PR.
> 
> The initial doc structure was inspired by [this website](https://documentation.divio.com/) and [PyCon talk](https://www.youtube.com/watch?v=t4vKPhjcMZg).
> 
> I have couple of questions based on the changes:
> 
> * `explanation` directory has been replaced with `design` directory. The initial idea was to include any sort of documentation that explains the theoretical foundations, including design principles and some algorithmic background. Do you think `design` directory would cover algorithmic background. Or do you think we wouldn't need to include algo background but just mention the design patterns here?
> * I see that `how-to-guides` is replaced with `additional-guides`. I wasn't hundred percent sure about `how-to-guides` anyway since it is not clearly distinguishable from `tutorials`. I'm now wondering whether it would be an idea to have `tutorials` section, which contains `beginner`, `intermediate` and `advanced` tutorials. We could place the `addiitonal-guides` to either `intermediate` or `advanced` tutorials. This way, user would know how to navigate between the `tutorials` and `additional-guides`. Also, the technical level of the tutorial would become more apparent. Any thoughts?
> * I am not sure if I understand the role of `samples` directory. Could you please elaborate?

I thought 'explanation' category is too abstract and 'how-to-guides' is like an implied meaning that gives a feeling of duplication with the tutorial. learning-oriented, problem-oriented, understanding-oriented, information-oriented were so clear, but the categories that imply them, I thought the expression was rather not clear. The category names have been changed, but the basic contents for them are the same you suggested. 'Design' can contain theoretical foundations, including some algorithmic background and design principles. 'additional-guides' meant extended guides to explain problem-oriented guides. Looks good that `tutorials` section has 3 different sub-categories like `beginner`, `intermediate` and `advanced`. It seems to be clearer and more intuitive navigation for users.

**Comment by chuneuny-emily** [comment]
> > @chuneuny-emily, thank you for creating this PR.
> > The initial doc structure was inspired by [this website](https://documentation.divio.com/) and [PyCon talk](https://www.youtube.com/watch?v=t4vKPhjcMZg).
> > I have couple of questions based on the changes:
> > 
> > * `explanation` directory has been replaced with `design` directory. The initial idea was to include any sort of documentation that explains the theoretical foundations, including design principles and some algorithmic background. Do you think `design` directory would cover algorithmic background. Or do you think we wouldn't need to include algo background but just mention the design patterns here?
> > * I see that `how-to-guides` is replaced with `additional-guides`. I wasn't hundred percent sure about `how-to-guides` anyway since it is not clearly distinguishable from `tutorials`. I'm now wondering whether it would be an idea to have `tutorials` section, which contains `beginner`, `intermediate` and `advanced` tutorials. We could place the `addiitonal-guides` to either `intermediate` or `advanced` tutorials. This way, user would know how to navigate between the `tutorials` and `additional-guides`. Also, the technical level of the tutorial would become more apparent. Any thoughts?
> > * I am not sure if I understand the role of `samples` directory. Could you please elaborate?
> 
> I thought 'explanation' category is too abstract and 'how-to-guides' is like an implied meaning that gives a feeling of duplication with the tutorial. learning-oriented, problem-oriented, understanding-oriented, information-oriented were so clear, but the categories that imply them, I thought the expression was rather not clear. The category names have been changed, but the basic contents for them are the same you suggested. 'Design' can contain theoretical foundations, including some algorithmic background and design principles. 'additional-guides' meant extended guides to explain problem-oriented guides. Looks good that `tutorials` section has 3 different sub-categories like `beginner`, `intermediate` and `advanced`. It seems to be clearer and more intuitive navigation for users.

About samples, I thought of code samples or Jupyter notebook examples that users modify with exportable code. @goodsong81 @samet-akcay @yunchu @kprokofi  What do you think of it?

File: docs/source/get_started/index.rst (hunk 1 above)
Comment (samet-akcay) [reply docs/source/get_started/index.rst:1]: yeah, looks like get started is more popular.
https://textranch.com/277981/getting-started/or/get-started/

**Comment by samet-akcay** [comment]
> About samples, I thought of code samples or Jupyter notebook examples that users modify with exportable code. @goodsong81 @samet-akcay @yunchu @kprokofi What do you think of it?

@chuneuny-emily, we could build the documentation with Jupyter notebooks that we store in the root directory. The advantages would be the following:

- The users would not need to search for Jupyter notebooks within the repo. They would be located in the main root.
- We could use these notebooks to build the notebooks in the documentation via [nbsphinx](https://nbsphinx.readthedocs.io/en/0.8.9/) extension.

For example, anomalib has notebooks directory in the root [here](https://github.com/openvinotoolkit/anomalib/tree/main/notebooks). The same notebooks are also used in the documentation as part of tutorials or how-to-guides. See the examples here [example1](https://openvinotoolkit.github.io/anomalib/how_to_guides/notebooks/100_datamodules/101_btech.html), [example2](https://openvinotoolkit.github.io/anomalib/how_to_guides/notebooks/200_models/201_fastflow.html), [example3](https://openvinotoolkit.github.io/anomalib/how_to_guides/notebooks/300_benchmarking/301_benchmarking.html).

**Comment by chuneuny-emily** [comment]
> > About samples, I thought of code samples or Jupyter notebook examples that users modify with exportable code. @goodsong81 @samet-akcay @yunchu @kprokofi What do you think of it?
> 
> @chuneuny-emily, we could build the documentation with Jupyter notebooks that we store in the root directory. The advantages would be the following:
> 
> * The users would not need to search for Jupyter notebooks within the repo. They would be located in the main root.
> * We could use these notebooks to build the notebooks in the documentation via [nbsphinx](https://nbsphinx.readthedocs.io/en/0.8.9/) extension.
> 
> For example, anomalib has notebooks directory in the root [here](https://github.com/openvinotoolkit/anomalib/tree/main/notebooks). The same notebooks are also used in the documentation as part of tutorials or how-to-guides. See the examples here [example1](https://openvinotoolkit.github.io/anomalib/how_to_guides/notebooks/100_datamodules/101_btech.html), [example2](https://openvinotoolkit.github.io/anomalib/how_to_guides/notebooks/200_models/201_fastflow.html), [example3](https://openvinotoolkit.github.io/anomalib/how_to_guides/notebooks/300_benchmarking/301_benchmarking.html).

Thanks for your good feedback, Samet. I removed sample category. We can use the notebooks as part of tutorials like you mentioned. Can you review the updated document structure?

**Comment by harimkang** [comment]
could you check this?
image

**Comment by kprokofi** [comment]
I think we can merge it for now to proceed development - and then update templates and content accordingly during implementation. @goodsong81 , @GalyaZalesskaya @samet-akcay  Do you agree?

**Review by goodsong81 — DISMISSED** [review]
@kprokofi Yeah, I think we can refine base on this PR.

**Comment by goodsong81** [comment]
Ah, we have code check failure...
image

**Comment by kprokofi** [comment]
@chuneuny-emily , Could you, please take a look at failure?

**Review by harimkang — APPROVED** [review]
LGTM,
I canceled the test not related to this PR.

**Comment by chuneuny-emily** [comment]
> @chuneuny-emily , Could you, please take a look at failure?

I fixed it and passed.

**Review by goodsong81 — APPROVED** [review]

**Pull request merged**